RxSwift/Observables/ObserveOn.swift

```diff
@@ -101,7 +101,36 @@ final fileprivate class ObserveOnSink<O: ObserverType> : ObserverBase<O.E> {
             _scheduleDisposable.disposable = self._scheduler.scheduleRecursive((), action: self.run)
         }
     }
-    
+
+    #if swift(>=3.2)
+    func run(_ state: Void, recurse: () -> Void) {
```

Comment:
Hi @alanzeino ,

I've merged this PR, but I've rearranged this part of code to always use Swift 3.2 version. Is there any reason why you've left the else part like it was? 
It seems to me that it's working this way.